.github/scripts/verifyActions.sh

```diff
@@ -8,16 +8,29 @@ declare -r GREEN='\033[0;32m'
 declare -r RED='\033[0;31m'
 declare -r NC='\033[0m'
 
+declare LIB_PATH
+LIB_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && cd ../../ && pwd)/node_modules/diff-so-fancy"
+
+# Expose diff-so-fancy to path
+PATH=$LIB_PATH:$PATH
```

Comment:
Are you sure this `LIB_PATH` stuff is even necessary? I don't think it is ... since it's added as a dev dependency, and we've already run `npm ci`, `diff-so-fancy` should already be available in the path.

You can test this locally by adding it as a dev dependency, running `npm i`, and then using `diff-so-fancy` in your terminal. It should work from the project root without needing to modify the PATH. I'm 95% sure `npm i` does that for you.

Reply:
Git needs the file inside the folder `diff-so-fancy` called by the same name. I find it very difficult that npm ci would place every folder in node_modules (individually) in the path as this would make all things collide. 

Did I understand you wrong?

Reply:
Check this out:

```
> diff-so-fancy
zsh: command not found: diff-so-fancy

> npm i --save-dev diff-so-fancy         
npm WARN @native-html/css-processor@1.6.1 requires a peer of @types/react-native@* but none is installed. You must install peer dependencies yourself.
npm WARN @native-html/css-processor@1.6.1 requires a peer of csstype@^3.0.4 but none is installed. You must install peer dependencies yourself.
npm WARN @native-html/transient-render-engine@3.6.1 requires a peer of @types/react-native@* but none is installed. You must install peer dependencies yourself.
npm WARN acorn-jsx@5.3.1 requires a peer of acorn@^6.0.0 || ^7.0.0 || ^8.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN detox@17.10.2 requires a peer of mocha@>=6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb@18.0.1 requires a peer of eslint@^5.16.0 || ^6.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb@18.0.1 requires a peer of eslint-plugin-react-hooks@^1.7.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb-base@14.0.0 requires a peer of eslint@^5.16.0 || ^6.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-import@2.20.0 requires a peer of eslint@2.x - 6.x but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-jsx-a11y@6.2.3 requires a peer of eslint@^3 || ^4 || ^5 || ^6 but none is installed. You must install peer dependencies yourself.
npm WARN react-native@0.63.3 requires a peer of react@16.13.1 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-config@1.4.1 requires a peer of react-native-windows@>=0.61 but none is installed. You must install peer dependencies yourself.
npm WARN tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none is installed. You must install peer dependencies yourself.
npm WARN The package @babel/preset-flow is included as both a dev and production dependency.

+ diff-so-fancy@1.3.0
added 2 packages from 3 contributors, updated 1 package and audited 2868 packages in 14.579s

136 packages are looking for funding
  run `npm fund` for details

found 26 vulnerabilities (17 low, 9 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details

> diff-so-fancy
Diff-so-fancy: https://github.com/so-fancy/diff-so-fancy
Version      : 1.3.0

Usage:

git diff --color | diff-so-fancy         # Use d-s-f on one diff
cat diff.txt | diff-so-fancy             # Use d-s-f on a diff/patch file
diff -u one.txt two.txt | diff-so-fancy  # Use d-s-f on unified diff output

diff-so-fancy --colors                   # View the commands to set the recommended colors
diff-so-fancy --set-defaults             # Configure git-diff to use diff-so-fancy and suggested colors

# Configure git to use d-s-f for *all* diff operations
git config --global core.pager "diff-so-fancy | less --tabs=4 -RFX"
```

So as you can see, I didn't have `diff-so-fancy` in my path, then I installed it using `npm i`, then after that I did have it in my path.

> I find it very difficult that npm ci would place every folder in node_modules (individually) in the path as this would make all things collide

I believe that it _does_ in fact place all of your dependencies in the PATH.

Reply:
hmm @roryabraham are you sure you don't have something specific in your bash configs that might be adding node_modules to your PATH?

checking out this branch locally, diff-so-fancy doesn't get added to my PATH after `npm i`
```
Expensidev/Expensify.cash (pr1726) $ npm i
npm WARN @native-html/css-processor@1.6.1 requires a peer of @types/react-native@* but none is installed. You must install peer dependencies yourself.
npm WARN @native-html/css-processor@1.6.1 requires a peer of csstype@^3.0.4 but none is installed. You must install peer dependencies yourself.
npm WARN @native-html/transient-render-engine@3.6.1 requires a peer of @types/react-native@* but none is installed. You must install peer dependencies yourself.
npm WARN acorn-jsx@5.3.1 requires a peer of acorn@^6.0.0 || ^7.0.0 || ^8.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN detox@17.10.2 requires a peer of mocha@>=6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb@18.0.1 requires a peer of eslint@^5.16.0 || ^6.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb@18.0.1 requires a peer of eslint-plugin-react-hooks@^1.7.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb-base@14.0.0 requires a peer of eslint@^5.16.0 || ^6.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-import@2.20.0 requires a peer of eslint@2.x - 6.x but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-jsx-a11y@6.2.3 requires a peer of eslint@^3 || ^4 || ^5 || ^6 but none is installed. You must install peer dependencies yourself.
npm WARN react-native@0.63.3 requires a peer of react@16.13.1 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-config@1.4.1 requires a peer of react-native-windows@>=0.61 but none is installed. You must install peer dependencies yourself.
npm WARN tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none is installed. You must install peer dependencies yourself.
npm WARN The package @babel/preset-flow is included as both a dev and production dependency.

added 43 packages from 24 contributors, removed 12 packages, updated 13 packages and audited 2867 packages in 16.43s

131 packages are looking for funding
  run `npm fund` for details

found 26 vulnerabilities (17 low, 9 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details


   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │      New major version of npm available! 6.14.9 → 7.6.3       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v7.6.3   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

Expensidev/Expensify.cash (pr1726) $ which diff-so-fancy
Expensidev/Expensify.cash (pr1726) $ diff-so-fancy
-bash: diff-so-fancy: command not found
```

but it's definitely installed
```
Expensidev/Expensify.cash (pr1726) $ ls -lah node_modules/ | grep diff
drwxr-xr-x     6 davidbondy  staff   192B Mar  2 13:40 arr-diff
drwxr-xr-x     7 davidbondy  staff   224B Mar  2 13:40 diff-sequences
drwxr-xr-x    19 davidbondy  staff   608B Mar 17 15:53 diff-so-fancy
drwxr-xr-x    10 davidbondy  staff   320B Mar  2 13:40 diffie-hellman
drwxr-xr-x     8 davidbondy  staff   256B Mar  2 13:40 fast-diff
drwxr-xr-x     6 davidbondy  staff   192B Mar  2 13:40 jest-diff
drwxr-xr-x     8 davidbondy  staff   256B Mar  2 13:40 semver-diff
Expensidev/Expensify.cash (pr1726) $ diff-so-fancy
-bash: diff-so-fancy: command not found
```

but I did find out some useful things from this so 
1. if we are able to edit the `git config` at the beginning of this file (i'm not sure if we are able to or not) then we can set `git config core.pager` to `diff-so-fancy` 
<img width="1024" alt="Screen Shot 2021-03-17 at 16 08 39" src="https://user-images.githubusercontent.com/4073354/111545676-09f87f00-873c-11eb-95fd-4a1066b27588.png">

2. BUT when you do that `git diff --exit-code` will not respect it and use the old pager 
<img width="1024" alt="Screen Shot 2021-03-17 at 16 10 40" src="https://user-images.githubusercontent.com/4073354/111545700-14b31400-873c-11eb-92a3-03945c2c9dbc.png">

so to skip worrying about if setting a git config actually works we can just bypass it and call the executable directly similarly to what you're doing now though I think we should invert how we're doing things a little.

So what I think the steps should be for this file are:
1. Set up PATH_TO_FANCY_DIFF_EXECUTABLE like you're doing with `LIB_PATH`
2. run `npm run gh-actions-build`
3. run `git diff --exit-code`
    1. if code is 0 then echo the success message and exit 0
    2. if code is > 0 then
        1. Echo the error message
        2. run `git diff | PATH_TO_FANCY_DIFF_EXECUTABLE | less --tabs=4 -RFX`
        3. exit 1

This way in "best case" runs we only run `git diff` once then in "worst case" runs we have to run it again to get the fancy diffing like we want in the output and we don't have to worry about juggling the error codes around from multiple calls to git diff.

Reply:
Hi @roryabraham  and @bondydaa ,

Oh, good things here. I didn't think about changing the pager config, thank you for pointing that out! I only put the diff-so-fancy in the path because that was what was recommended in the docs and I was not sure that if I called it directly I would be inviting trouble in the future with anything.

But since you are aware of it now, I am happy to edit the code to have a directly call to the `LIB_PATH` as `PATH_TO_FANCY_DIFF_EXECUTABLE` but I just can't guarantee it will run smooth always.

Reply:
Ah interesting, lets go with just calling the executable for now and if we see weird behavior we can change it to just add it to the path

Reply:
Hi @bondydaasorry for the late reply. I thought I had already replied before! I understand, the required changes were done.